src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -687,6 +706,7 @@ public virtual void OnDeserialization(object? sender)
             int realVersion = siInfo.GetInt32(VersionName);
             int hashsize = siInfo.GetInt32(HashSizeName);
             _comparer = (IEqualityComparer<TKey>)siInfo.GetValue(ComparerName, typeof(IEqualityComparer<TKey>))!; // When serialized if comparer is null, we use the default.
+            FixupComparer();
```

Comment:
The binary serialization does not preserve object identity of singletons, so this is going to be no-op as far as I can tell.

What is not going work or work differently if this `FixupComparer` call is deleted here?

Reply:
You're right, it doesn't do anything. I have removed the call (and inlined back the remaining invocation of the method).